gk/main.c

```diff
@@ -2215,12 +2215,12 @@ get_responsible_gk_mailbox(const struct ip_flow *flow,
 	uint16_t queue_id;
 	int block_idx;
 
-	/*
-	 * XXX #149 Change the mapping rss hash value to rss reta entry
-	 * if the reta size is not 128.
-	 */
-	RTE_VERIFY(gk_conf->rss_conf_front.reta_size == 128);
-	rss_hash_val = (rss_hash_val & 127);
+	RTE_VERIFY(gk_conf->rss_conf_front.reta_size > 0);
+	if (rte_is_power_of_2(gk_conf->rss_conf_front.reta_size)) {
+		rss_hash_val = (rss_hash_val &
+			(gk_conf->rss_conf_front.reta_size - 1));
+	} else
+		rss_hash_val = rss_hash_val % gk_conf->rss_conf_front.reta_size;
```

Comment:
I see the good intention in testing if `gk_conf->rss_conf_front.reta_size` is a power of 2, but I think that the added overhead is about the same cost of having `rss_hash_val = rss_hash_val % gk_conf->rss_conf_front.reta_size;` for all cases.